### PR TITLE
SpecAugment-like time and frequency masks

### PR DESCRIPTION
**Pull request by vsl9**
Added time and frequency masks similar to SpecAugment (https://arxiv.org/abs/1904.08779)

### CODE REVIEW

File: open_seq2seq/data/speech2text/speech_utils.py [hunk 1]
```diff
@@ -399,6 +400,22 @@ def get_speech_features_librosa(signal, sample_freq, num_features,
...
+    width_freq_mask = augmentation.get('width_freq_mask', 10)
+    width_time_mask = augmentation.get('width_time_mask', 50)
+
+    for idx in range(n_freq_mask):
+      freq_band = np.random.randint(width_freq_mask + 1)
+      freq_base = np.random.randint(0, features.shape[1] - freq_band)
+      features[:, freq_base:freq_base+freq_band] = 0
+    for idx in range(n_time_mask):
+      time_band = np.random.randint(width_time_mask + 1)
+      if features.shape[0] - time_band > 0:
```
Comment (blisc) [inline open_seq2seq/data/speech2text/speech_utils.py]: Do we need a similar check for freq_band?
Shouldn't we also add an else that drops the entire spectrogram?

Comment (vsl9) [reply open_seq2seq/data/speech2text/speech_utils.py]: Frequency shape is fixed (that is, `num_audio_features`). Time duration varies from utterance to utterance. I'd vote for skipping masking in that case.

**Comment by vadimkantorov** [comment]
The original [SpecAugment paper](https://arxiv.org/pdf/1904.08779.pdf) uses much higher values for parameters: `width_freq_mask = 27` (number of filter banks they use is 80; jasper uses 6 with 64 filter banks) and `width_time_mask = 100` (jasper uses 6)

The Google's SpecAugment code https://github.com/tensorflow/lingvo/blob/master/lingvo/core/spectrum_augmenter.py#L37-L42 uses less aggressive values: `width_freq_mask = 10` and `width_time_mask = 50`

Could you please comment if you tried more aggressive values (compared to 6)? Thanks!

**Comment by vadimkantorov** [comment]
Hmm, though Google's SpecAugment code seems to use `freq_mask_count = 1` and `time_mask_count = 1`, so all in all the difference is probably not very significant

**Pull request merged**